### ISSUE TITLE
Texture/UVmapping problems

### ISSUE THREAD

**Issue by drtyhlpr** [google/filament#708]
Thanks for the great library!

I had some hard texture mapping problems with my first tries to import basic 3D models via obj files and display them on Android in the style of textured-object.(https://github.com/google/filament/tree/master/android/samples)

So my first goal was to create a simple as possible object in blender, UV unwrap it and use it in a simple as possible sample project.

**Steps to reproduce the behavior**
1. Create OBJ in Blender (Add Torus with Generate UVs. Add 1024x1024 Color Grid test image/texture.png)
2. Export Wavefont .obj (tried different settings: include normals, keep vertex order etc.)
3. Successfully open(ed) output.obj with open3mod or meshlab -> looks good
4. Use **textured-object** sample project and use it with the generated model and texture with simple **baked_texture** material.
5. Successfully build an run the app in emulator

**Expected behavior**
I excepted the object to be shown in similar quality like in open3mod or meshlab.

**Experienced behavior**
The Mesh is displayed correctly but the texture mapping has many errors. The top part of the Object (relative to Y axis) mostly has no mapping at all.

**Ideas**
Well I am a little bit stuck here....

- Is this problem related to https://github.com/google/filament/issues/672 ?
- The default shader_ball obj seems to be generated from a blender object (are there any special settings in blender to enable? - I tried many different .obj export configurations)
- Is the object (uv data) converted incorrectly by **filamesh** ? (verbose output and or error checks would be great) 

ps. Software and library used latest versions of filament, android studio and blender

**Comment by bejado** [comment]
Can you post a comparison screenshot between Filament and what you expect? I don't think this is related to #672 but you could flip the Y axis in your material to quickly verify.

**Comment by romainguy** [comment]
I use Blender myself all the time either to import or create models and I haven't experienced specific problems. It could very well be a UV flip issue.

**Comment by drtyhlpr** [comment]
Hi, thanks for your ultra fast replys

Picture of texture mapping problem:
![problem](https://i.imgur.com/41WxN7C.png)

**Comment by romainguy** [comment]
Wow, that is very bizarre indeed. Mind sending us the .obj file you are using?

**Comment by drtyhlpr** [comment]
No problem. Give me some minutes to find a place

**Comment by drtyhlpr** [comment]
I have uploaded the blender, obj and texture image to:
https://github.com/drtyhlpr/filament-stuff

**Comment by drtyhlpr** [comment]
Thanks for your info and screenshot. I had problems building material_sanbox and windows and currently don't have another OS available. I hope to have material_sanbox working soon heh

I started a new Android Project fully based on https://github.com/google/filament/tree/master/android/samples/textured-object
using
https://github.com/google/filament/releases/download/sceneform-1.7pr2/filament-20181213-android.aar

Expect the changed object and adding the texture to the project I only changed the material to baked_texture and changed/reduced the setupMaterial() function to something like that:
```
        val sampler = TextureSampler();
        baseColor = loadTexture(engine, resources, R.drawable.torustexture, TextureType.COLOR)
        materialInstance.setParameter("albedo", baseColor, sampler)
```

**Comment by romainguy** [comment]
I can confirm it looks wrong. I'll take a closer look.

**Comment by drtyhlpr** [comment]
puh thank you now I feel a little less stupid (working all day on this). I fixed the vs2017 building issues I had and waiting for llightbulb/material sandbox compilation... thanks for your help and this great lib this is real 3D porn for me.

**Comment by romainguy** [comment]
You can check the fix in PR #709. It works fine with your test model now, sorry about this and thanks for the report!

**Comment by drtyhlpr** [comment]
Thank you!
